vscode/src/qirGeneration.ts

```diff
@@ -46,19 +51,23 @@ export async function getQirForActiveWindow(): Promise<string> {
       setTarget("base");
     }
   }
-
-  // Get the diagnostics for the current document.
-  const diagnostics = await vscode.languages.getDiagnostics(
-    editor.document.uri,
-  );
-  if (diagnostics?.length > 0) {
-    throw new QirGenerationError(
-      "The current program contains errors that must be fixed before submitting to Azure",
-    );
+  let sources: [string, string][] = [];
+  try {
+    sources = await loadProject(editor.document.uri);
+    for (const source of sources) {
+      const diagnostics = await vscode.languages.getDiagnostics(
+        vscode.Uri.parse(source[0]),
+      );
+      if (diagnostics?.length > 0) {
+        throw new QirGenerationError(
+          "The current program contains errors that must be fixed before submitting to Azure",
+        );
+      }
+    }
+  } catch (e: any) {
+    throw new QirGenerationError(e.message);
```

Comment:
This `try/catch` should probably be directly around `loadProject` since that's the actual thing it seems to be intended to catch. Otherwise we might be catching and rethrowing the exception we just threw in line 63... which feels a bit unnecessary.